### PR TITLE
feat(HAC-5814): allow users to set context in UI

### PR DESCRIPTION
**Pull request by CryptoRodeo**
## Fixes 
[HAC-5814](https://issues.redhat.com/browse/HAC-5814)

## Description
Allow users to set and edit the `contexts`  field of IntegrationTestScenarios in the Konflux UI.
Users can choose to add or remove contexts from a predefined list of supported contexts.
The `application` context is set as the default context.

## Type of change

- [x] Feature
- [ ] Bugfix
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Screen shots / Gifs for design review 

**Before**:
image

**After**:
image

image

image

## Browser conformance: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: src/components/IntegrationTest/ContextsField.tsx [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+      } else if (!isSelected) {
+        // Select, add necessary data
+        arrayHelpers.push({ name: contextName, description: currentContext.description });
+      }
+    },
+    [contexts, selectedContexts],
+  );
+
+  return (
+    <div>
```
Comment (sahil143) [inline src/components/IntegrationTest/ContextsField.tsx:145]: use FormGroup Component from Patternfly to set the label and fieldId.

**Review by sahil143 — CHANGES_REQUESTED** [review]
A couple of changes required 

- Move Contexts before the Parameters field.
- Applications context should be selected by default when user opens the create form.

**Comment by CryptoRodeo** [comment]
I have applied the following updates:
- Move contexts above Parameters field
- Pre-select the 'applications' context when creating a new integration test

see changes here: https://github.com/openshift/hac-dev/pull/997/commits/8681758ee1f7a6f481d09fde572b39500903533e

I will get to the FormGroup suggestion soon and squash all commits after.

File: src/components/IntegrationTest/ContextsField.tsx (hunk 1 above)
Comment (CryptoRodeo) [reply src/components/IntegrationTest/ContextsField.tsx]: FormGroup added with label and field id: https://github.com/openshift/hac-dev/pull/997/files#diff-f46d00580c9b2d98a2c8088a4544bea1d84ff334dd4914835883c3ab97dfc358R154

**Comment by CryptoRodeo** [comment]
Moved back to draft. It appears that we should use a typeahead instead of a regular select drop down: https://issues.redhat.com/browse/CWFCONF-10814?focusedId=25900474&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-25900474

**Review by dirgim — APPROVED** [review]
LGTM

**Comment by caugello** [comment]
@CryptoRodeo all settings are displayed in overview. Should this not be included as well ?

**Comment by CryptoRodeo** [comment]
@caugello Correct, it is included in this MR: https://github.com/openshift/hac-dev/pull/1008

The main changes are here: https://github.com/openshift/hac-dev/pull/1008/commits/7c68163fc09b67cf9c30fcb00825bd380db277fe

**Review by caugello — APPROVED** [review]

**Pull request merged**